AStarAlgorithmsProject/Alg Scripts/AStarSolver.cs

```diff
@@ -0,0 +1,231 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Diagnostics;
+
+namespace AStarAlgorithmsProject
+{
+    class AStarSolver
+    {
+        enum TileStates { Unchecked, Open, Closed }; // enum to title and track the 3 states a tile can be in.
+        static Point[] Directions = new Point[] { new Point(-1, 1), new Point(0, 1), new Point(1, 1),
+                                                  new Point(-1, 0), new Point(1, 0),
+                                                  new Point(-1, -1), new Point(0, -1), new Point(1, -1)
+                                                }; // in order: NW, N, NE, W, E, SW, S, SE
+
+        private static int MapSize; // the total NxN size of the map being worked with.
+
+        private Tile[,] map; // the map A* will traverse and the tiles therin
+        private List<Tile> openTiles; // the queue of tiles being evaluated to find the desired path.
+
+        public AStarSolver(int mS) // default constructor where mS is the number of rows/columns in the map
+        {
+            map = new Tile[mS, mS];
+            MapSize = mS;
+            InitMap();
+        }
+
+        // "Prime" the map so all tiles are in their default state and their locations set to the proper cordinates in the map
+        private void InitMap()
+        {
+            for (int i = 0; i < MapSize; i++)
+            {
+                for (int j = 0; j < MapSize; j++)
+                {
+                    map[i, j] = new Tile(new Point(i, j));
+                }
+            }
+        }
+
+        // Sets the a tile at location l to the value of p
+        public void SetPassalbe(Point l, bool p)
+        {
+            map[l.X, l.Y].Passable = p;
+        }
+
+        // Sets the scalar cost of a tile at point l to the value of c
+        public void SetMovementCost(Point l, int c)
+        {
+            map[l.X, l.Y].CostScalar = c;
+        }
+
+        // Starting at the goal, traced back through the parents of each tile until it reachest the start tile, and returns a 
+        // list of the points on the path in the order of start -> finish
+        public List<Point> GetPath(Point s, Point g)
+        {
+            List<Point> p = new List<Point>();
+            
+            if (!(ValidTile(s) && ValidTile(g)))
+            {
+                throw new ArgumentException("What the fuck is this shit you little bitch why are you giving point parameters out of bounds of the map size I should have you drawn, quartered, then redrawn so you don't look quite so fucking stupid");
```

Comment:
Please change this

Reply:
I am sorry. One of those jokes not even funny to the person who made it. Won't happen again.